### PR TITLE
Mini cart

### PR DESCRIPTION
**Pull request by peter-gerhard**
@lauraluiz @schleichardt please review

### CODE REVIEW

File: app/cart/LineItemAddController.java [hunk 1]
```diff
@@ -29,9 +30,13 @@ public LineItemAddController(final ControllerDependency controllerDependency) {
             final ProductVariantToCartFormData data = form.get();
             final F.Promise<Cart> cartPromise = getOrCreateCart(userContext, session());
             return cartPromise.flatMap(cart -> {
-                final AddLineItem action = AddLineItem.of(data.getProductId(), data.getVariantId(), ObjectUtils.firstNonNull(data.getAmount(), 1));
+                final Long itemCount = ObjectUtils.firstNonNull(data.getAmount(), 1L);
```
Comment (schleichardt) [inline app/cart/LineItemAddController.java]: I would prefer a static import for `ObjectUtils.firstNonNull`.

File: common/app/common/pages/SunrisePageDataFactory.java [hunk 2]
```diff
@@ -44,7 +45,13 @@ private PageHeader getPageHeader(final PageContent pageContent) {
         final String title = messages.at("header.title") + pageContent.additionalTitle();
         final CollectionData<SelectableData> countries = getCountries();
         final NavMenuData navMenuData = getNavMenuData();
-        return new PageHeader(messages, title, countries, navMenuData);
+        final MiniCartData miniCartData = getMiniCartData();
+        return new PageHeader(messages, title, countries, navMenuData, miniCartData);
+    }
+
+    private MiniCartData getMiniCartData() {
+        final String url = String.format("/%s/cart", userContext.locale().toLanguageTag());
```
Comment (schleichardt) [inline common/app/common/pages/SunrisePageDataFactory.java]: The `ReverseRouter` is not available here?

File: common/app/common/cart/MiniCartActions.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    public static void decreaseCartItemCount(final Long itemCount) {
+        increaseCartItemCount(-itemCount);
+    }
+
+    public static void increaseCartItemCount(final Long itemCount) {
+        setCartItemCount(getCartItemCount() + itemCount);
+    }
+
+    public static void setCartItemCount(final Long itemCount) {
+        session().put(CartSessionKeys.CART_ITEM_COUNT, String.valueOf(itemCount));
```
Comment (schleichardt) [inline common/app/common/cart/MiniCartActions.java]: I think it would improve the design if you need the session as parameter.

**Comment by peter-gerhard** [comment]
@schleichardt can you take a second look, please :)

**Comment by schleichardt** [comment]
:+1:

**Pull request merged**